acinclude/nettle.m4

```diff
@@ -15,8 +15,8 @@ AC_DEFUN([SQUID_CHECK_NETTLE_BASE64],[
   AH_TEMPLATE(HAVE_NETTLE34_BASE64,[set to 1 if Nettle 3.4 API will link])
   SQUID_STATE_SAVE(squid_nettle_base64_state)
   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#   include <cinttypes>
 #   include <cstddef>
-#   include <cstdint>
```

Comment:
> This line is being touched ensure that the configure test is checking things work with the same includes that Squid itself uses later for the actual build.

We should either rely on "intypes.h includes stdint.h" guarantee or never use that guarantee. If, starting with this PR, we rely on that guarantee, then this change should be undone[^1] because the existing configure test is already "checking things work with the same includes that Squid itself uses later for the actual build". For that "actual build", Squid will explicitly include cinttypes, but we know that Squid will "use" cstdint as well!

[^1]: Assuming this code is using the correct header in the first place, of course. If this code actually needs cinttypes, then this change is not wrong, just out of this PR scope. My change request is based on the quoted rationale. I have not checked whether this code uses any cinttypes APIs that are not provided by cstdint.

IMO, we should not tell folks to include B.h instead of A.h when the standard tells them that the API they use is defined in A.h. Two changes in this PR, including this one, imply the opposite. I am against such changes and consider their stated rationale contradicting the guarantee that this PR is based on.